lib/model/model.go

```diff
@@ -2306,26 +2407,98 @@ func (m *model) AddConnection(conn protocol.Connection, hello protocol.Hello) {
 	}
 
 	m.deviceWasSeen(deviceID)
+	m.scheduleConnectionPromotion()
+}
+
+func (m *model) scheduleConnectionPromotion() {
+	m.promotionTimer.Reset(time.Second)
```

Comment:
Well... It's a one second schedule after adding or removing connections. The reason for doing this is that if a device shuts down and we have 16 connections to it, we will get 16 connection closes in rapid succession. We don't want/need to run the promotion process 16 times during that time, but we do want to run it once things have quieted down. I think if the connection frequency is such that "after one second" becomes "never" then there's a different problem to be solved...

As for AfterFunc vs in serve, I think it's nice to have it called in one place under known circumstances, instead of spread out / scheduled in places where it's not visible. Plus we'd need to track the timer returned by AfterFunc to be able to cancel it and renew, which would need locking and whatnot...

Reply:
Oh I totally didn't get that it's a feature that it keeps deferring the execution, not an unfortunate side-effect.  That definitely makes the `AfterFunc` point entirely moot. Eternal delay... Yeah I guess we can ignore that, even with tons of devices that really shouldn't be a thing (also some bad connection cycle should still be >>1s).

I'll give it another shot at proposing a comment, lets see if this turns out comprehensible (and without missed gotchas) :) 

```suggestion
	// Keeps deferring to prevent multiple executions in quick succession, e.g. if multiple connections to a single device are closed.
	m.promotionTimer.Reset(time.Second)
```